apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -134,14 +136,33 @@ SOMAExperimentAxisQuery <- R6::R6Class(
         assert_subset(obs_column_names, self$obs_df$colnames(), "column"),
         "'var_column_names' must be a character vector" =
           is.null(var_column_names) || is.character(var_column_names),
-        assert_subset(var_column_names, self$var_df$colnames(), "column")
+        assert_subset(var_column_names, self$var_df$colnames(), "column"),
+        "'n_cores' must be a single positive integer value" = rlang::is_integerish(n_cores, 1L) &&
+          n_cores > 0L
       )
 
+      if (Sys.info()['sysname'] == 'Windows') {
+        if (n_cores != 1L) {
+          warning(
+            "Parallelization is not supported on Windows",
```

Comment:
It's been years for me since I mucked with Windows but as I recall even package `parallel` supports a version of parallelism via `mclapply` -- see the help.  There are some limits because `fork` isn't there.

But the bigger point may be that this is a non-issue for us:  `tiledbsoma` is a non-Windows package right now.  I think we should keep it simpler and remove all Windows concerns because 

https://github.com/single-cell-data/TileDB-SOMA/blob/797ac6720e4fdbf257e29d337cde851806a732e2/apis/r/DESCRIPTION#L67

Reply:
I misremembered: Windows 'tolerates' `mclapply` but falls back to serial `lapply` use essentially already giving us the fallback made explicit here.  

So for simplicitly (and while there is no Windows build) I still think we can remove the special case made here that will simply not be triggered until we have a Windows build.